pyro/util.py

```diff
@@ -207,9 +208,14 @@ def enum_extend(trace, msg, num_samples=None):
         num_samples = -1
 
     # Batched .enumerate_support() assumes batched values are independent.
-    shape = broadcast_shape(msg["fn"].shape(*msg["args"], **msg["kwargs"]),
-                            msg["value"].size())
-    batch_dims = len(shape) - msg['fn'].event_dim(*msg["args"], **msg["kwargs"])
+    if isinstance(msg["fn"], RandomPrimitive):
```

Comment:
Wow I admit to creating this logic, and I see it is gross. I'll try to remove it asap.

Reply:
I was using it myself in the tests, but fortunately, Pyro internals otherwise look fairly decoupled.